x-pack/plugins/maps/public/classes/sources/esql_source/esql_source.tsx

```diff
@@ -57,12 +57,17 @@ export class ESQLSource extends AbstractVectorSource implements IVectorSource {
       throw new Error('Cannot create ESQLSourceDescriptor when esql is not provided');
     }
 
+    if (!isValidStringConfig(descriptor.dataViewId)) {
```

Comment:
Have ES|QL layers in Maps been promoted to serverless yet? In a local instance I created a Map with an ES|QL layer from a previous build of Kibana. As such, this layer did not have a `dataViewId` which crashed the map on load. 

If ES|QL map layers are not in serverless yet, then I think this is fine.

Reply:
> Have ES|QL layers in Maps been promoted to serverless yet?
not sure. 

> In a local instance I had previously created a Map with an ES|QL layer from a previous build of Kibana. As such, this layer did not have a dataViewId which crashed the map on load in this PR.

This issue is recored in  https://github.com/elastic/kibana/issues/176318. I am working on a fix now. Its not complicated, add try/catch to all usages of `createLayerInstance`. In redux action `addLayer` display toast that layer could not be added when `createLayerInstance` throws

> If ES|QL map layers are not in serverless yet, then I think this is fine.

I think once the bug is fixed, then its ok even if this has been released in serverless. Serverless is still in beta and ES|QL layers are also in beta (double beta)! I think potentially displaying an error toast for a few users is better then an ugly work around in the code base. What do you think?

Reply:
I'm ok with the changes since we've already identified #176318 as a bug that needs to be fixed.

"double beta" 😆

Reply:
I will wait until a fix for https://github.com/elastic/kibana/issues/176318 is merged before merging this. That way, there will be no chance that this PR gets included in serverless without a fix to the identified problem.